src/Elmah/XmlFileErrorLog.cs

```diff
@@ -158,10 +173,36 @@ public override string Log(Error error)
                 writer.Flush();
             }
 
+            DeleteOldFiles();
             return errorId;
         }
 
         /// <summary>
+        /// Deletes Old files according the <code>size</code> parameter
+        /// </summary>
+        private void DeleteOldFiles()
+        {
+            try
+            {
+                if (!Directory.Exists(LogPath))
```

Comment:
You can [use `DirectoryInfo` to make an existence check](https://msdn.microsoft.com/en-us/library/system.io.directoryinfo.exists(v=vs.110).aspx) as well, as in:

```c#
var dir = new DirectoryInfo(LogPath);
if (!dir.Exists)
    return;
```